tests/test_sharded_zarr_store.py

```
@@ -0,0 +1,669 @@
+            await uninitialized_store._load_root_from_cid()
+
+        # Test dimensionality mismatch
+        assert store._parse_chunk_key("temp/c/0/0/0") is None  # 3D key for 2D array
+
+        # Test invalid coordinates
+        assert store._parse_chunk_key("temp/c/3/0") is None  # Out of bounds (3 >= 2 chunks)
+        assert store._parse_chunk_key("temp/c/0/invalid") is None  # Non-integer
+        assert store._parse_chunk_key("temp/c/0/-1") is None  # Negative coordinate
+
+@pytest.mark.asyncio
+async def test_sharded_zarr_store_init_invalid_shapes(create_ipfs: tuple[str, str]):
```

Ruff (0.11.9)

Redefinition of unused `test_sharded_zarr_store_init_invalid_shapes` from line 490

(F811)